core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2541,6 +2541,16 @@ private[spark] object Utils extends Logging {
     master == "local" || master.startsWith("local[")
   }
 
+  /**
+   * Push based shuffle can only be enabled when external shuffle service is enabled.
+   * In the initial version, we cannot support pushed based shuffle and adaptive execution
+   * at the same time. Will improve this in a later version.
+   */
+  def isPushBasedShuffleEnabled(conf: SparkConf): Boolean = {
+    conf.get(PUSH_BASED_SHUFFLE_ENABLED) && conf.get(SHUFFLE_SERVICE_ENABLED) &&
+      !conf.getBoolean("spark.sql.adaptive.enabled", false)
```

Comment:
AQE no longer affects push based shuffle.

Reply:
@venkata91 You need to cherry-pick this `60a8750997` from magnet-upstream